ncloud/resource_ncloud_nks_cluster.go

```diff
@@ -0,0 +1,354 @@
+				Computed: true,
+			},
+			"acg_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Required:         true,
+				ValidateDiagFunc: ToDiagFunc(validation.StringLenBetween(3, 30)),
+			},
+			"cluster_type": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: ToDiagFunc(validation.StringInSlice([]string{"SVR.VNKS.STAND.C002.M008.NET.SSD.B050.G002", "SVR.VNKS.STAND.C004.M016.NET.SSD.B050.G002"}, false)),
+				Required:         true,
+				ForceNew:         true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"region_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"login_key_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"k8s_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpc_no": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vpc_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_no_list": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"subnet_lb_no": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"subnet_lb_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"log": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"audit": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNcloudNKSClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*ProviderConfig)
+	if !config.SupportVPC {
+		return diag.FromErr(NotSupportClassic("resource `ncloud_nks_cluster`"))
+	}
+
+	reqParams := &vnks.ClusterInputBody{
+		RegionCode: &config.RegionCode,
+		//Required
+		Name:         StringPtrOrNil(d.GetOk("name")),
+		ClusterType:  StringPtrOrNil(d.GetOk("cluster_type")),
+		LoginKeyName: StringPtrOrNil(d.GetOk("login_key_name")),
+		K8sVersion:   StringPtrOrNil(d.GetOk("k8s_version")),
+		ZoneCode:     StringPtrOrNil(d.GetOk("zone")),
+		VpcNo:        getInt32FromString(d, "vpc_no"),
+		SubnetLbNo:   getInt32FromString(d, "subnet_lb_no"),
+	}
+
+	if list, ok := d.GetOk("subnet_no_list"); ok {
+		reqParams.SubnetNoList = expandSubnetNoList(list.([]interface{}))
```

Comment:
```suggestion
		reqParams.SubnetNoList = expandStringInterfaceList(list.([]interface{}))
```

기존에 있는 `expandStringInterfaceList` 를 활용하시면 좋을 것 같습니다.

Reply:
NKS SDK에 Subnet No List가 []*int32로 생성되어 string을 사용하는 다른 리소스와 다르게 expandStringInterfaceList 사용이 어렵습니다.